google-cloud-core/src/main/java/com/google/cloud/HttpTransportOptions.java

```diff
@@ -154,10 +155,26 @@ public void initialize(HttpRequest httpRequest) throws IOException {
         if (readTimeout >= 0) {
           httpRequest.setReadTimeout(readTimeout);
         }
+
+        HttpHeaders headers = httpRequest.getHeaders();
+        headers.set("x-goog-api-client", getXGoogApiClientHeader());
       }
     };
   }
 
+  String getXGoogApiClientHeader() {
+    return String.format(
+        "gl-java/%s %s/%s",
```

Comment:
Relying on the absence of a key is a very shaky strategy... we never know what variants we will cook up in the future, and none of them will be distinguishable when no key is present to disambiguate them. I feel it is quite important to know old-platform vs old-platform+gapic vs new-platform+grpc vs new-platform+http. I almost feel like that is more important than `gccl`.